packages/node/src/node/client/legacy/InteractionClient.ts

```diff
@@ -0,0 +1,1136 @@
+                              .map(
+                                  ({ endpointId, clusterId, dataVersion }) =>
+                                      `${endpointId}/${clusterId}=${dataVersion}`,
+                              )
+                              .join(", ")
+                        : undefined,
+                    eventFilters: eventFilters?.length
+                        ? eventFilters.map(({ nodeId, eventMin }) => `${nodeId}=${eventMin}`).join(", ")
+                        : undefined,
+                    fabricFiltered: isFabricFiltered,
+                    keepSubscriptions,
+                    minInterval: Duration.format(Seconds(minIntervalFloorSeconds)),
+                    maxInterval: Duration.format(Seconds(maxIntervalCeilingSeconds)),
+                }),
+        });
+
+        const report = fullReport;
+        fullReport = undefined; // Prevent further data collection and handle updates
+
+        updateReceived?.();
+
+        return {
+            ...report,
+            maxInterval: subscribe.maxInterval,
+        };
+    }
+
+    async #invoke<C extends Command<any, any, any>>(
+        options: InvokeOptions<C> & { suppressResponse: boolean },
+    ): Promise<ResponseType<C>> {
+        const { command: commandObj } = options;
+
+        const {
+            endpointId,
+            clusterId,
+            command: { requestId, requestSchema, timed },
+            asTimedRequest,
+            timedRequestTimeout: timedRequestTimeoutMs = DEFAULT_TIMED_REQUEST_TIMEOUT,
+            expectedProcessingTime,
+            useExtendedFailSafeMessageResponseTimeout = false,
+            skipValidation,
+        } = options;
+        let { suppressResponse, request } = options;
+        const timedRequest =
+            (timed && !skipValidation) || asTimedRequest === true || options.timedRequestTimeout !== undefined;
+
+        if (this.isGroupAddress) {
+            if (endpointId !== undefined) {
+                throw new ImplementationError("Invoking a concrete command on a group address is not supported.");
+            }
+            if (timedRequest) {
+                throw new ImplementationError("Timed requests are not supported for group address invokes.");
+            }
+            suppressResponse = true;
+        }
+
+        if (requestSchema instanceof ObjectSchema) {
+            if (request === undefined) {
+                // If developer did not provide a request object, create an empty one if it needs to be an object
+                // This can happen when all object properties are optional
+                request = {} as RequestType<C>;
+            }
+            if (requestSchema.isFabricScoped && request.fabricIndex === undefined) {
+                request.fabricIndex = FabricIndex.NO_FABRIC;
+            }
+        }
+
+        if (!skipValidation) {
+            requestSchema.validate(request);
+        }
+        const commandFields = requestSchema.encodeTlv(request);
```

Comment:
Rather than encoding yourself I think you can just provide the TLV and other parameters in an `Invoke` and let `ClientInteraction` take care of this for you with its existing interface.  `Invoke` is designed in terms of "bare" matter values + TLV schema such as you have here

Reply:
ahh yeeahh me blond... sorry